src/test/java/com/wikia/webdriver/testcases/discussions/ReportingPostTests.java

```diff
@@ -0,0 +1,620 @@
+package com.wikia.webdriver.testcases.discussions;
+
+import com.wikia.webdriver.common.contentpatterns.MercuryWikis;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.core.annotations.Execute;
+import com.wikia.webdriver.common.core.annotations.InBrowser;
+import com.wikia.webdriver.common.core.drivers.Browser;
+import com.wikia.webdriver.common.core.helpers.Emulator;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.PostEntity;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.PostsCreator;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.Transitions;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PageWithPosts;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostDetailsPage;
+import com.wikia.webdriver.elements.mercury.pages.discussions.PostsListPage;
+import com.wikia.webdriver.elements.mercury.pages.discussions.UserPostsPage;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Execute(onWikia = MercuryWikis.DISCUSSIONS_AUTO)
+@Test(groups = "discussions-reporting-posts")
+public class ReportingPostTests extends NewTestTemplate {
+
+  private static final String DESKTOP_RESOLUTION = "1920x1080";
+
+  private final String POST_ON_DESKTOP_ON_POSTS_LIST = "post-desktop-posts-list";
+
+  private final String POST_ON_DESKTOP_ON_POST_DETAILS = "post-desktop-post-details";
+
+  private final String POST_ON_DESKTOP_ON_USER_PAGE = "post-desktop-user-page";
+
+  private final String POST_ON_MOBILE_ON_POSTS_LIST = "post-mobile-posts-list";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this field "POST_ON_MOBILE_ON_POSTS_LIST" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this final field static too. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1170)